backends/graphics3d/openglsdl/openglsdl-graphics3d.cpp

```diff
@@ -632,6 +664,17 @@ OpenGL::FrameBuffer *OpenGLSdlGraphics3dManager::createFramebuffer(uint width, u
 }
 
 void OpenGLSdlGraphics3dManager::updateScreen() {
+#if defined(USE_IMGUI) && SDL_VERSION_ATLEAST(2, 0, 0)
+	ImGui_ImplOpenGL3_NewFrame();
+	ImGui_ImplSDL2_NewFrame(_window->getSDLWindow());
+
+	ImGui::NewFrame();
+	_metaEngine->renderImGui();
```

Comment:
What happenswhen no engine is running? E.g. in GUI?

What happens after returning to the launcher? Will it try calling already freed _metaEngine?

What happens when the next game is launched?

In general, this design is error-prone, and it seems to me that the backend code will allocate the whole engine now, so we will have two copies (?), or at least, two references.

I am not yet sure how to resolve it, perhaps, by enforcing calling `updateImGui()` from the engine code as a compromise? Similar to how we do with our debugger. Or, creating ImGui as a singleton system-wide?

Reply:
Yes, I know, I was expecting a comment on this. It was a proposition and I was expecting a solution from one of you as I'm just a newbie in ScummVM 😉 .
I have to think about it...